x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/feature/SimpleFeatureFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.vectortile.feature;
+
+import com.wdtinc.mapbox_vector_tile.VectorTile;
+import com.wdtinc.mapbox_vector_tile.encoding.GeomCmd;
+import com.wdtinc.mapbox_vector_tile.encoding.GeomCmdHdr;
+import org.apache.lucene.util.BitUtil;
+import org.elasticsearch.geometry.Rectangle;
+
+/**
+ * Similar to {@link FeatureFactory} but only supports points and rectangles. It is just
+ * much more efficient for those shapes.
+ */
+public class SimpleFeatureFactory {
+
+    private final int extent;
+    private final Rectangle rectangle;
+    private final double pointXScale, pointYScale;
+
+    public SimpleFeatureFactory(int z, int x, int y, int extent) {
+        this.extent = extent;
+        rectangle = FeatureFactoryUtils.getTileBounds(z, x, y);
+        pointXScale = 1d / ((rectangle.getMaxLon() - rectangle.getMinLon()) / (double) extent);
```

Comment:
is there a reason not to do this simpler expression instead? If so please add a comment. :)
```suggestion
        pointXScale = (double) extent / (rectangle.getMaxLon() - rectangle.getMinLon());
```